Edit docs CI workflows to accommodate Docusaurus

ptgott opened the pull request:
- Copy the `gravitational/teleport` source into the `content` directory of a `gravitational/docs-website` clone and overwrite the `git-update` yarn script with a no-op command. With way, we can use the `gravitational/teleport` clone across multiple workflow steps (e.g., the prose linting step), and can identify a branch to pull using the `actions/checkout` defaults. Name the submodule directory `current` to match the expectations of the Docusaurus site.

- Remove the Vercel preview workflow. After the docs engine migration, we'll no longer need to deploy a preview to Vercel in order to show docs authors what their potential changes look like on a rendered docs site.

Also fixes spelling errors caught by a later version of cspell.

ptgott committed:
Edit docs CI workflows to accommodate Docusaurus

- Copy the `gravitational/teleport` source into the `content` directory
  of a `gravitational/docs-website` clone and overwrite the `git-update`
  yarn script with a no-op command. With way, we can use the
  `gravitational/teleport` clone across multiple workflow steps (e.g.,
  the prose linting step), and can identify a branch to pull using the
  `actions/checkout` defaults. Name the submodule directory `current` to
  match the expectations of the Docusaurus site.

- Remove the Vercel preview workflow. After the docs engine migration,
  we'll no longer need to deploy a preview to Vercel in order to show
  docs authors what their potential changes look like on a rendered docs
  site.

Also fixes spelling errors caught by a later version of cspell.

zmb3 commented on .github/workflows/doc-tests.yaml:
```diff
@@ -44,8 +44,14 @@ jobs:
         with:
-          repository: "gravitational/docs"
-          path: "docs"
+          repository: 'gravitational/teleport'
+          path: 'teleport'
```
Wouldn't it be equivalent to omit these two lines now?

ptgott commented:
Converting to draft until we're closer to making the final switch.

fheinecke commented on .github/workflows/doc-tests.yaml:
```diff
@@ -80,30 +80,36 @@ jobs:
+        #
+        # We also replace data fetched from Sanity CMS with hardcoded JSON
+        # objects to remove the need to authenticate with Sanity. Each includes
+        # the minimal set of data required for docs builds to succeed.
```
Please run this step through shellcheck. There's several issues that it will report.

ptgott committed:
Fix shellcheck issues in doc-tests.yaml script

- Add `exit` commands for failed `cd`s
- Don't read from and write to `package.json` in the same pipeline
- Add double quotes around variables

ptgott committed:
Use latest Docusaurus site config schema

ptgott committed:
Fix the Prepare docs site configuration step

- Remove unneeded `cd`s
- Use the correct version name in the `config.json` prepared for the
  Docusaurus site

ptgott commented:
@fheinecke Checking to see if you have time to give this one another look. Thanks!

taraspos commented:
@ptgott we may want to remove vercel webhook as well:
- https://github.com/gravitational/teleport/blob/074d584b7a9447dfde15314455bf7e82f3604ccf/.github/workflows/update-docs-webhook.yaml#L20-L21

Unless we want to keep it for couple days and then possibly remove it next week

ptgott commented:
> @ptgott we may want to remove vercel webhook as well:
> 
> * https://github.com/gravitational/teleport/blob/074d584b7a9447dfde15314455bf7e82f3604ccf/.github/workflows/update-docs-webhook.yaml#L20-L21
> 
> Unless we want to keep it for couple days and then possibly remove it next week

Thanks! I'll plan to remove this in a separate PR since we can run `Lint (Docs)` using `gravitational/docs-website` even before we route `/doc` traffic to the new site.
